### DISCUSSION TITLE
help - shivammathur/php not working on Catalina

### DISCUSSION THREAD

**Discussion by akagumo** [shivammathur/homebrew-php#952] [Q&A / Help] [answered]
I have been using php 7.3 built-in and apache build-in on Catalina fine, but Drupal now requires newer so I installed homebrew shivammathur/php.  I installed php@8.1 following all the steps, terminal show all installed correctly.  The browser shows "can't connect to server" when trying to run any php.

So I installed php@7.4 and linked and updated httpd.conf and php runs kind of.  But when trying run Drupal update php hangs. So still a problem.  I also tried php@8.0, same as php@8.1, it does not work.

How do get this to work?

I followed these instructions https://wpbeaches.com/updating-to-php-versions-7-4-and-8-on-macos-11-big-sur-and-catalina/
which seem the same as these instructions https://github.com/shivammathur/homebrew-php/blob/master/README.md

**Comment by shivammathur** [comment, marked as answer]
Please follow this guide. It would work on Catalina as well. Choose the processor as intel in the tip at the beginning.
https://getgrav.org/blog/macos-monterey-apache-multiple-php-versions

**Reply by akagumo** [reply]
there is no way to use the built-in apache with the homebrew php?  I have done all the steps except setup homebrew httpd.

**Reply by shivammathur** [reply]
It should work, but I would recommend using apache from `brew`.

**Reply by akagumo** [reply]
the above link seems to have several errors in the post.

the module location is not correct
#LoadModule php_module /opt/homebrew/opt/php@8.0/lib/httpd/modules/libphp.so
instead the below location works.
LoadModule php_module /usr/local/opt/php@8.1/lib/httpd/modules/libphp.so

also this is not where php.ini is located.
/opt/homebrew/etc/php/7.0/php.ini
this is actual path:
local/etc/php/8.1/php.ini

**Reply by shivammathur** [reply]
In the post, select intel processor, then follow it

![image](https://user-images.githubusercontent.com/1571086/179976072-0370eae8-4321-431e-a043-52e2795b958f.png)

**Reply by akagumo** [reply]
oops. I guess when the page reloaded the next day I missed that it set itself back to apple silicon.
